lib/net.js

```diff
@@ -709,12 +706,9 @@ protoGetter('remotePort', function remotePort() {
 Socket.prototype._getsockname = function() {
   if (!this._handle || !this._handle.getsockname) {
     return {};
-  }
-  if (!this._sockname) {
-    const out = {};
-    const err = this._handle.getsockname(out);
-    if (err) return {};  // FIXME(bnoordhuis) Throw?
-    this._sockname = out;
+  } else if (!this._sockname) {
+    // FIXME(bnoordhuis) Throws when returns not 0?
+    this._handle.getsockname(this._sockname = {});
```

Comment:
Please assign the value in a separate line. That way it's easier to follow the code.